### PR TITLE
Fix docs for decoder_input_ids in BART.

### LINKED ISSUE #15691: About `decoder_input_ids` in BART doc
> For translation and summarization training, decoder_input_ids should be provided. If no decoder_input_ids is provided, the model will create this tensor by shifting the input_ids to the right for denoising pre-training following the paper.
https://huggingface.co/docs/transformers/v4.16.2/en/model_doc/bart#transformers.BartForConditionalGeneration.forward.decoder_input_ids

This is inappropriate. If `labels` is provided, the model will create this tensor by shifting the `labels`:
https://github.com/huggingface/transformers/blob/db7d6a80e82d66127b2a44b6e3382969fdc8b207/src/transformers/models/bart/modeling_bart.py#L1320-L1324

### PR DESCRIPTION
**Pull request by ShaneTian**
Fix docs for `decoder_input_ids` in BART.
Fix #15691

### CODE REVIEW

File: src/transformers/models/bart/modeling_tf_bart.py [hunk 1]
```diff
@@ -575,8 +575,11 @@ def serving(self, inputs):
             is used, optionally only the last `decoder_input_ids` have to be input (see `past_key_values`).
 
             For translation and summarization training, `decoder_input_ids` should be provided. If no
-            `decoder_input_ids` is provided, the model will create this tensor by shifting the `input_ids` to the right
-            for denoising pre-training following the paper.
+            `decoder_input_ids` is provided, the model will automatically create this tensor by shifting:
+
+            - the `labels` to the right for text generation if `labels` is provided,
+            - the `input_ids` to the right for denoising pre-training following the paper if no `labels` is provided.
+
```
Comment (NielsRogge) [inline src/transformers/models/bart/modeling_tf_bart.py:582]:
```suggestion
            `decoder_input_ids` is provided, the model will automatically create this tensor by shifting:

            - the `labels` to the right for text generation if `labels` are provided,
            - the `input_ids` to the right for denoising pre-training following the paper if no `labels` are provided.
```suggestion
            `decoder_input_ids` is provided, the model will automatically create this tensor by shifting:

            - the `labels` to the right for text generation if `labels` are provided,
            - the `input_ids` to the right for denoising pre-training following the paper if no `labels` are provided.

```
```

File: src/transformers/models/bart/modeling_bart.py [hunk 2]
```diff
@@ -599,8 +599,11 @@ def __init_subclass__(self):
             is used, optionally only the last `decoder_input_ids` have to be input (see `past_key_values`).
 
             For translation and summarization training, `decoder_input_ids` should be provided. If no
-            `decoder_input_ids` is provided, the model will create this tensor by shifting the `input_ids` to the right
-            for denoising pre-training following the paper.
+            `decoder_input_ids` is provided, the model will automatically create this tensor by shifting:
+
+            - the `labels` to the right for text generation if `labels` is provided,
+            - the `input_ids` to the right for denoising pre-training following the paper if no `labels` is provided.
```
Comment (NielsRogge) [inline src/transformers/models/bart/modeling_bart.py:605]:
```suggestion
            `decoder_input_ids` are provided, the model will automatically create this tensor by shifting:

            - the `labels` to the right for text generation if `labels` are provided,
            - the `input_ids` to the right for denoising pre-training following the paper if no `labels` are provided.
```
```suggestion
            `decoder_input_ids` is provided, the model will automatically create this tensor by shifting:

            - the `labels` to the right for text generation if `labels` is provided,
            - the `input_ids` to the right for denoising pre-training following the paper if no `labels` is provided.
```

**Review by NielsRogge — COMMENTED** [review]
Thanks for fixing!

Can you also run `make fixup` locally in a virtual environment?

**Comment by NielsRogge** [comment]
Thanks! However, as you can see, the CI needs all checks passing before we can merge this PR.

For this to pass, you need to run `make fixup` locally and make sure all quality and style checks pass.

File: src/transformers/models/bart/modeling_bart.py [hunk 3]
```diff
@@ -599,8 +599,11 @@ def __init_subclass__(self):
             is used, optionally only the last `decoder_input_ids` have to be input (see `past_key_values`).
 
             For translation and summarization training, `decoder_input_ids` should be provided. If no
-            `decoder_input_ids` is provided, the model will create this tensor by shifting the `input_ids` to the right
-            for denoising pre-training following the paper.
+            `decoder_input_ids` are provided, the model will automatically create this tensor by shifting:
+
+            - the `labels` to the right for text generation if `labels` are provided,
+            - the `input_ids` to the right for denoising pre-training following the paper if no `labels` are provided.
+
```
Comment (patil-suraj) [inline src/transformers/models/bart/modeling_bart.py:606]: I'm not sure if this statement is correct. BART is pre-trained with sentence-permutation and text-infilling objective. Where the input text is noised and is passed to the encoder and the decoder is tasked to generate the original document. See this figure from the paper.
Screenshot 2022-02-21 at 12 16 06 PM

So for this task it's not possible to create `decoder_input_ids` by shifting `input_ids`.

This is only True for the sequence-classification task where we feed the same input to both encoder and decoder.
see this figure from the paper: 
Screenshot 2022-02-21 at 12 16 24 PM

cc @patrickvonplaten
